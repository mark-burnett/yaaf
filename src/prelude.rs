pub use crate::{
    actor::{Actor, Tell},
    context::{Context, Publish},
    handler::Handler,
    source::Source,
    system::System,
};
pub use ::async_trait::async_trait;
pub use ::yaaf_macros::{Actor, Source};
