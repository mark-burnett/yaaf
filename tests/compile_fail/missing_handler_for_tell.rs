use ::yaaf::prelude::*;

#[derive(Clone, Debug)]
struct ValidMessage;

#[derive(Clone, Debug)]
struct InvalidMessage;

#[derive(Actor)]
#[handle(ValidMessage)]
struct MyActor;

#[async_trait]
impl Handler<ValidMessage> for MyActor {
    async fn handle(&mut self, _ctx: &mut Context<Self>, _message: ValidMessage) {
    }
}

#[tokio::main]
async fn main() -> Result<(), Box<dyn ::std::error::Error>> {
    let mut system = System::new().await?;

    let actor = MyActor;

    let address = system.add_actor(actor).await?;

    address.tell(InvalidMessage)?;

    Ok(())
}
