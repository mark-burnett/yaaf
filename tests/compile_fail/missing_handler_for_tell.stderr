error[E0308]: mismatched types
  --> $DIR/missing_handler_for_tell.rs:27:18
   |
27 |     address.tell(InvalidMessage)?;
   |                  ^^^^^^^^^^^^^^ expected struct `ValidMessage`, found struct `InvalidMessage`
